File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/carts/domain/usecase/GetCartUseCase.java
package com.tarbadev.witchcraft.carts.domain.usecase;

import com.tarbadev.witchcraft.carts.domain.entity.Cart;
import com.tarbadev.witchcraft.carts.domain.repository.CartRepository;
import org.springframework.stereotype.Component;

@Component
public class GetCartUseCase {
  private final CartRepository cartRepository;

  public GetCartUseCase(CartRepository cartRepository) {
    this.cartRepository = cartRepository;
  }

  public Cart execute(int id) {
    return cartRepository.findById(id);
  }
}

File: WitchcraftApp/src/main/kotlin/com/tarbadev/witchcraft/carts/domain/usecase/GetCartUseCase.kt
package com.tarbadev.witchcraft.carts.domain.usecase

import com.tarbadev.witchcraft.carts.domain.entity.Cart
import com.tarbadev.witchcraft.carts.domain.repository.CartRepository
import org.springframework.stereotype.Component

@Component
class GetCartUseCase(private val cartRepository: CartRepository) {

    fun execute(id: Int): Cart? {
        return cartRepository.findById(id)
    }
}
